### PR TITLE
Refactored the router and routes code

### PR DESCRIPTION
**Pull request by sandlerben**

### CODE REVIEW

**Comment by sandlerben** [comment]
r @yoninachmany

File: main.go [hunk 1]
```diff
@@ -1,60 +1,15 @@
...
-
-	r.HandleFunc("/add_job", initiateTranscriptionJobHandler)
-
-	// add middleware
-	stderrLoggingHandler := func(http.Handler) http.Handler {
-		return handlers.LoggingHandler(os.Stderr, r)
-	}
-	middlewareRouter := alice.New(handlers.CompressHandler, stderrLoggingHandler).Then(r)
+	router := NewRouter()
+	middlewareRouter := ApplyMiddleware(router)
```
Comment (yoninachmany) [inline main.go]: screen shot 2016-02-18 at 3 52 10 pm
ideas why?

File: router.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"github.com/justinas/alice"
+)
+
+// NewRouter creates and returns a mux.Router with default routes
+func NewRouter() *mux.Router {
+	router := mux.NewRouter()
+	router.HandleFunc("/hello/{name}", helloHandler)
+	router.HandleFunc("/add_job", initiateTranscriptionJobHandler)
```
Comment (yoninachmany) [inline router.go]: should you delete these `HandleFunc`s if they exist in `routes.go`?

File: router.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+			Methods(route.Method).
+			Path(route.Pattern).
+			Name(route.Name).
+			Handler(route.HandlerFunc)
+	}
+
+	return router
+}
+
+// ApplyMiddleware wraps the router in some middleware
```
Comment (yoninachmany) [inline router.go]: in comment, maybe add more specific description of middleware? also please add period - "Doc comments work best as complete sentences..."

File: routes.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+		"GET",
+		"/hello/{name}",
+		helloHandler,
+	},
+	route{
+		"initiateTranscriptionJob",
+		"DELETE",
+		"/add_job",
+		initiateTranscriptionJobHandler,
+	},
```
Comment (yoninachmany) [inline routes.go:34]: don't forget about `{"health", "GET", "/health", healthHandler}`

also, make sure to add 
`func healthHandler(w http.ResponseWriter, r *http.Request) {
    w.Write([]byte("healthy!"))
 }`

File: router.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import (
+	"net/http"
+	"os"
+
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+	"github.com/justinas/alice"
+)
+
+// NewRouter creates and returns a mux.Router with default routes
```
Comment (yoninachmany) [inline router.go]: what are "default routes"? also please add period - "Doc comments work best as complete sentences..."

File: router.go (hunk 2 above)
Comment (sandlerben) [inline router.go]: yes, good catch.

File: router.go (hunk 3 above)
Comment (sandlerben) [inline router.go]: :+1:

File: routes.go (hunk 4 above)
Comment (sandlerben) [inline routes.go:34]: yes, when I pull I can do that

**Comment by sandlerben** [comment]
@yoninachmany look good now?

File: routes.go (hunk 4 above)
Comment (yoninachmany) [inline routes.go:34]: please explain? health is already merged in; pull from where?

Comment (sandlerben) [inline routes.go:34]: exactly, I already merged it

Comment (yoninachmany) [inline routes.go:34]: meant that health is merged into master. as long as the method gets included, lgtm

Comment (sandlerben) [inline routes.go:34]: health was merged into master earlier this week

**Pull request merged**